### DISCUSSION TITLE
Split sinks-elasticsearch feature into sinks-elasticsearch and sinks-elasticsearch-aws

### DISCUSSION THREAD

**Discussion by suikammd** [vectordotdev/vector#18269] [General]
Currently, the sinks-elasticsearch feature in Vector enables AWS support by default, which may not be necessary for all users. This leads to an increase in compiled code size for users who only require the normal Elasticsearch functionality. To address this, I propose splitting the sinks-elasticsearch feature into two separate features: sinks-elasticsearch and sinks-elasticsearch-aws.

Or we can make sinks-elasticsearch feature disable aws on default and enable aws when aws-core feature enabled? 

Another idea is add sinks-elasticsearch-common feature for normal Elasticsearch, this won't introduce breakchange. Like the following:
`
sinks-elasticsearch-common = ["transforms-metric_to_log"]
sinks-elasticsearch = ["sinks-elasticsearch-common", "aws-core"]
`

I can help make it if you wish to make a change :)

**Comment by jszwedko** [comment]
Hi @suikammd !

Thanks for raising this up. I agree that it would make sense to feature flag the AWS bits of the Elasticsearch sink. We have a few other components that also include AWS bits that aren't always needed so I think having a general flag, like `aws-core`, that they could all use would make sense. Happy to see a PR for that!

**Reply by suikammd** [reply]
Thanks for your relay!
I want to make sure you know this way would introduce breakchange. If you are already aware of that, I will start to do this job.

**Reply by jszwedko** [reply]
It would, but only for users compiling Vector themselves, which we don't really offer compatibility guarantees around at the moment. The distributed version of Vector will, of course, have this flag enabled.

**Reply by suikammd** [reply]
hi @jszwedko could you help review this pr? thx~
https://github.com/vectordotdev/vector/pull/18288
